app/components-react/windows/IconLibraryProperties.tsx

```diff
@@ -0,0 +1,111 @@
+import cx from 'classnames';
+import path from 'path';
+import fs from 'fs';
+import { ModalLayout } from '../shared/ModalLayout';
+import { $t } from '../../services/i18n';
+import React, { useEffect, useState } from 'react';
+import { Services } from '../service-provider';
+import { FileInput } from '../shared/inputs';
+import styles from './IconLibraryProperties.m.less';
+
+export default () => {
+  // inject services
+  const { SourcesService, WindowsService, CustomizationService } = Services;
+
+  const sourceId = WindowsService.getChildWindowQueryParams().sourceId;
+  const source = SourcesService.views.getSource(sourceId);
+
+  const [folderPath, setFolderPath] = useState('');
+  const [folderImages, setFolderImages] = useState(['']);
+  const [selectedIcon, setSelectedIcon] = useState('');
+
+  useEffect(lifecycle, []);
+
+  function lifecycle() {
+    if (source) {
+      const settings = source.getPropertiesManagerSettings();
+      if (settings.folder) {
+        const { folder, activeIcon } = settings;
+        setFolderPath(folder);
+
+        fs.readdir(folder, (err: Error, files: string[]) => {
+          setFolderImages(files.map((file: string) => path.join(folder, file)));
```

Comment:
Should handle the error case here.  When accessing the filesystem, errors are common and to be expected.